src/v/pandaproxy/schema_registry/types.h

```diff
@@ -489,3 +489,44 @@ from_string_view<compatibility_level>(std::string_view sv) {
 }
 
 } // namespace pandaproxy::schema_registry
+
+template<>
+struct fmt::formatter<pandaproxy::schema_registry::schema_reference> {
+    constexpr auto parse(fmt::format_parse_context& ctx)
+      -> decltype(ctx.begin()) {
+        auto it = ctx.begin();
+        auto end = ctx.end();
+        if (it != end && (*it == 'l' || *it == 'e')) {
+            presentation = *it++;
+        }
+        if (it != end && *it != '}') {
+            throw fmt::format_error("invalid format");
+        }
+        return it;
+    }
+
+    template<typename FormatContext>
+    auto format(
+      const pandaproxy::schema_registry::schema_reference& s,
+      FormatContext& ctx) const -> decltype(ctx.out()) {
+        if (presentation == 'l') {
+            return fmt::format_to(
+              ctx.out(),
+              "name: {}, subject: {}, version: {}",
+              s.name,
+              s.sub,
+              s.version);
+        } else {
+            return fmt::format_to(
+              ctx.out(),
+              "name='{}', subject='{}', version={}",
+              s.name,
+              s.sub,
+              s.version);
+        }
+    }
+
+    // l : format for logging
+    // e : format for error_reporting
```

Comment:
> This is nice. How come we don't just use the error_reporting for logging too though?

Because it's noisy and repetitive.

Reply:
Aren't they almost the same except for the format `{key}: {value}` vs `{key}='{value}'`? Am I missing something here?

Reply:
> Aren't they almost the same except for the format `{key}: {value}` vs `{key}='{value}'`? Am I missing something here?

The error message has the schema, with it's type and references, and then repeats the references. I don't want to log that duplication.

Reply:
I meant that we could use the same way of formatting `schema_reference`'s in both logs (with `operator<<`) and in the error message (`no_reference_found_for`). Something like this https://github.com/redpanda-data/redpanda/commit/07ab2d290e1fbde7d9f2c7f38a36066b6d021ea1. I don't see how that would cause logging the references twice.

It's not a big deal. I am just wondering if I am missing something here.

Reply:
Oh, right, I get it, sorry, yeah. Mostly just consistency with all the other log messages from SR.